projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product-carousel/product-carousel.e2e.cy.ts

```diff
@@ -99,31 +103,28 @@ context('Product carousel', () => {
             method: 'GET',
             pathname: `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
               'BASE_SITE'
-            )}/products/${secureDigitalCard2gb}`,
+            )}/products/search`,
           },
-          secureDigitalCardRequestHandler
-        );
+          searchRequestHandler
+        ).as('searchProducts');
 
-        function clickNextButton(): void {
-          const cypressButton = cy.get(
-            'cx-product-carousel:first-of-type button.next'
-          );
+        visitHomepage();
 
-          cypressButton.then((button) => {
-            if (!button.is('[aria-disabled="true"]')) {
-              cypressButton.click();
-              clickNextButton();
-            }
-          });
-        }
+        cy.wait('@searchProducts'); // Verify the call was made
 
-        // Continue clicking next in the carousel until we are at the last slide.
-        clickNextButton();
+        // Verify that the Add to Cart button is visible at least for the first item
+        const firstItemAddToCartButton = cy
+          .get('cx-product-carousel:first-of-type cx-product-carousel-item')
+          .first()
+          .find('cx-add-to-cart button');
+        firstItemAddToCartButton.should('be.visible');
 
-        const lastSlide = cy
+        // Verify that the Add to Cart button is not visible for the item that is out of stock (the last item):
+        const lastItemAddToCartButton = cy
           .get('cx-product-carousel:first-of-type cx-product-carousel-item')
-          .last();
-        lastSlide.find('cx-add-to-cart button').should('not.exist');
```

Comment:
Note for the reviewer:
By the way, while running e2e of this file, I noticed the previous implementation of this test wasn't testing the feature properly:
- all the 'Add to cart' buttons were appearing only after a delay in Cypress tests, so the assertion "last AddToCart button should not exist" was always truthy
  - now we "wait until the first AddToCart button appears". and only then "check  that the last AddToCartbutton should not exist"
- Since 9 months the carousel doesn't use `/product` endpoint, but `/products/search`, but this e2e test wasn't updated (i.e. the mocked API wasn't changed). It wasn't noticed before likely because the test passed due to the flaw of the delayed appearing of buttons with negative assertion ("should not exist")